### ISSUE TITLE
System theme at install wizard not found

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#4956]
### Expected behavior

- database installed
- open http://friendica.local
- show wizard

### Actual behavior

```
[Mon Apr 30 08:06:04.321621 2018] [:error] [pid 27296] [client 192.168.22.1:58100] PHP Fatal error:  Uncaught Exception: No system theme config value set. in /vagrant/src/App.php:1122\nStack trace:\n#0 /vagrant/src/App.php(1107): Friendica\\App->computeCurrentTheme()\n#1 /vagrant/index.php(325): Friendica\\App->getCurrentTheme()\n#2 {main}\n  thrown in /vagrant/src/App.php on line 1122
[Mon Apr 30 08:06:31.592900 2018] [:error] [pid 27293] [client 192.168.22.1:58137] PHP Fatal error:  Uncaught Exception: No system theme config value set. in /vagrant/src/App.php:1122\nStack trace:\n#0 /vagrant/src/App.php(1107): Friendica\\App->computeCurrentTheme()\n#1 /vagrant/index.php(325): Friendica\\App->getCurrentTheme()\n#2 {main}\n  thrown in /vagrant/src/App.php on line 1122
[Mon Apr 30 08:11:29.661217 2018] [:error] [pid 27301] [client 192.168.22.1:58422] PHP Fatal error:  Uncaught Exception: No system theme config value set. in /vagrant/src/App.php:1122\nStack trace:\n#0 /vagrant/src/App.php(1107): Friendica\\App->computeCurrentTheme()\n#1 /vagrant/index.php(325): Friendica\\App->getCurrentTheme()\n#2 {main}\n  thrown in /vagrant/src/App.php on line 1122
[Mon Apr 30 08:12:06.229092 2018] [:error] [pid 27295] [client 192.168.22.1:58461] PHP Fatal error:  Uncaught Exception: No system theme config value set. in /vagrant/src/App.php:1122\nStack trace:\n#0 /vagrant/src/App.php(1107): Friendica\\App->computeCurrentTheme()\n#1 /vagrant/index.php(325): Friendica\\App->getCurrentTheme()\n#2 {main}\n  thrown in /vagrant/src/App.php on line 1122
```

### Steps to reproduce the problem

- Use the `Vagrantfile` for the environment
- Drop database & recreate it
- delete `.htconfig.php`
- Go to `http:192.168.22.10`

### Friendica version you encountered the problem

2018-05-dev

### Friendica source (git, zip)

https://github.com/friendica/friendica

### PHP version

```
PHP 7.0.28-0ubuntu0.16.04.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.28-0ubuntu0.16.04.1, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans
```

### SQL version

`mysql  Ver 14.14 Distrib 5.7.22, for Linux (x86_64) using  EditLine wrapper`

**Comment by tobiasd** [comment]
What are you trying to do?

The Vagrant VM is for developers who don't want/can setup a Friendica node to test their work on.

**Comment by nupplaphil** [comment]
I try to re-test the installation for my dev-branch :-)

But nevertheless if I drop&recreate the db and delete `.htconfig.php` the install-wizard should appear.

**Comment by tobiasd** [comment]
It should yes :-) I usually start with a fresh VM to test the installer to leave the Vagrant VM as it is with the test data.

**Comment by MrPetovan** [comment]
The Exception has been added in #4951

Previously, if there was no system theme config setting, we were scanning the theme folder and taking the first one as the system theme. This weird fallback has been removed and now requires that a config value be set.

Is this affecting regular install as well?

**Comment by nupplaphil** [comment]
I believe it does.

This is out of a "normal" clone:
> Fatal error: Uncaught Exception: No system theme config value set. in /var/www/html/src/App.php:1122 Stack trace: #0 /var/www/html/src/App.php(1107): Friendica\App->computeCurrentTheme() #1 /var/www/html/index.php(325): Friendica\App->getCurrentTheme() #2 {main} thrown in /var/www/html/src/App.php on line 1122

I think there has to be a "fallback" in case there is no `.htconfig.php`

**Comment by MrPetovan** [comment]
I'm reluctant to hardcode a value in the code, I'd like us to have a default config file that could be loaded before an eventual custom `.htconfig.php` file. But first we have to have a specific `config` folder (#4889). I won't be able to work on it until the end of the month though, but if you can that'd be awesome.

Plus you get to choose the file format of the new configuration files since we're dropping the global `$a`. I'm partial for Neon (https://ne-on.org/) but you get to make the decision if you decide to work on it.

**Comment by astifter** [comment]
@nupplaphil Yep, it still does:
```
2018/05/04 23:05:05 [error] 14424#14424: *3 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Exception: No system theme config value set. in /home/astifter/friendica/src/App.php:1122
Stack trace:
#0 /home/astifter/friendica/src/App.php(1107): Friendica\App->computeCurrentTheme()
#1 /home/astifter/friendica/index.php(325): Friendica\App->getCurrentTheme()
#2 {main}
  thrown in /home/astifter/friendica/src/App.php on line 1122" while reading response header from upstream, client: 192.168.56.1, server: friendica-local.astifter.com, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.0-fpm.sock:", host: "friendica-local.astifter.com"
```
Any quick fixes for that?

**Comment by MrPetovan** [comment]
Apart installing with master and then upgrading to develop, no.

**Comment by astifter** [comment]
@MrPetovan ah, yes, true.

**Comment by nupplaphil** [comment]
@mrpetovan so we would have two config files:
- standard config file for default/fallback usage
- custom config file which "override" some/all of the standard config

I'm currently working on my docker files, but when I'm done I'll think about it :-)

**Comment by MrPetovan** [comment]
Yeah, the default config file could also double as in place documentation about the file-only config values.

**Comment by MrPetovan** [comment]
Closed via #5091
